numpy/core/tests/test_records.py

```diff
@@ -416,6 +416,21 @@ def test_pickle_3(self):
             assert_(pa.flags.writeable)
             assert_(pa.flags.aligned)
 
+    def test_pickle_void(self):
+        # issue gh-13593
+        dt = np.dtype([('obj', 'O')])
+        a = np.empty(1, dtype=dt)
+        data = (bytearray(b'eman'),)
+        a['obj'] = data
+        state = a[0].__reduce__()
+        # If the data is bytes, it will pickle the address of the content
+        # which cannot be saved/restored
+        assert not isinstance(state[1][1], bytes)
+        b = state[0](*state[1])
+        assert b[0] == (bytearray(b'eman'),)
+        # got a fresh version, not the original pointers rebuilt
+        assert b[0] is not data
+
```

Comment:
A round-trip pickling check would require careful deletion of `data` to make sure it was pickling properly